### PR TITLE
doc: Add RELEASE_PLANS.md

### PR DESCRIPTION
**Pull request by ahuber21**
I propose to add a new file: `RELEASE_PLANS.md` in which we collect actionable items and other TODOs for upcoming releases. My other PR #2460 deprecates some model builder functions in favor of new versions with more parameters. This file will help us to keep track of such changes.

### CODE REVIEW

**Comment by KulikovNikita** [comment]
Looks interesting to me

File: RELEASE_PLANS.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Upcoming releases: Plans and required actions
```
Comment (aepanchi) [inline RELEASE_PLANS.md:1]:
```suggestion
# Upcoming Releases: Plans and Required Actions
```

**Review by aepanchi — APPROVED** [review]
small suggestion, the rest looks good

**Comment by napetrov** [comment]
Talked offline but also would put this here.
The intent of this change is to have forcing function for cleaning up deprecated functionality - but i don't see how this implementation would help, other than adding another place with deprecation mention.

Also from  terminology perspective
deprecation - is fact of declaring deprecation that something will be removed
removal - is actually fact of removing functionality/api

So we have to declare deprecation in:
1. release notes
2. deprecated functionality section in documentation
3. potentially add deprecated mechanics in code

From removal perspective this have to be an activity planned for next major release to complete cleanup.

Also important considerations - deprecating functionality in code and comments should be done without implying versions. i.e. deprecated functionality is not enforcing timelines or targets.

**Comment by ahuber21** [comment]
Thank you Nikolay, these points make sense to me. Closing this PR as we'll be taking a different route.

**Pull request closed without merging**